### PR TITLE
Production dependencies

### PR DESCRIPTION
**Pull request by stuartmcalpine**

### CODE REVIEW

**Comment by stuartmcalpine** [comment]
- The dependencies table now has `input_production_id` and `input_id`, when registering an execution, there as now both `input_ids=[]` and `input_production_ids=[]` lists
- Had to modify the database creation script to use `declarative_base` to create foreign keys between schemas. Now the production schema is created alongside the default one (though it can be ignored with `no_production` flag in args.
- `TableCreator` no longer gets used, its all in `create_registry_db.py`

**Review by JoanneBogart — APPROVED** [review]
Basically seems fine. Please take a look at the minor comments I made, one requiring action (to fix a typo); the other maybe not.   No need for re-review.

File: scripts/create_registry_db.py [hunk 1]
```diff
@@ -3,144 +3,326 @@
...
+Both schemas have the same layout, containing six tables:
+    - "dataset"         : Primary table, contains information on the datasets
+    - "dataset_alias"   : Table to associate "alias" names to datasets
+    - "execution"       : Stores the executions, datasets can be linked to these
+    - "execution_alias" : Table to asscociate "alias" names to executions
+    - "dependancy"      : Tracks dependencies between datasets
+    - "provenance"      : Contains information about the database/schema 
+"""
+
+Base = declarative_base()
```
[the remark this answers is not in the document]
Comment (stuartmcalpine) [reply scripts/create_registry_db.py]: I have changed it

File: tests/end_to_end_tests/create_test_entries.py [hunk 2]
```diff
@@ -37,9 +37,7 @@
 
 # Establish connection to production database (if not sqllite)
 if datareg.db_connection.dialect != "sqllite":
-    datareg_prod = DataRegistry(
-        root_dir=_TEST_ROOT_DIR_PRODUCTION, schema="production"
-    )
+    datareg_prod = DataRegistry(root_dir=_TEST_ROOT_DIR_PRODUCTION, schema="production")
```
[the remark this answers is not in the document]
Comment (stuartmcalpine) [reply tests/end_to_end_tests/create_test_entries.py:40]: Thanks!, I always say sql-lite in my head!

**Pull request merged**